src/clients/FtpGateway.cc

```diff
@@ -704,18 +702,22 @@ ftpListParseParts(const char *buf, struct Ftp::GatewayFlags flags)
                 p->size = atoi(ct + 1);
                 break;
 
-            case 'm':
-                tm = (time_t) strtol(ct + 1, &tmp, 0);
-
-                if (tmp != ct + 1)
+            case 'm': {
+                char *tmp;
+                const auto tm = (time_t) strtol(ct + 1, &tmp, 0);
```

Comment:
This is poorly specified, but [AFAICT](https://datatracker.ietf.org/doc/html/draft-bernstein-eplf-02), we do not want to accept hex and octal integers in this context:

```
      "m"[time]
         This file was last modified at [time]. [time] is a sequence of
         ASCII digits specifying a number of seconds, real time, since
         the beginning of 1970 GMT. This fact cannot be used for files
         modified before 1970 GMT.
```

We also should reject negative timestamps instead of risking converting them to positive time_t, right?

```suggestion
                const auto seconds = strtol(ct + 1, &tmp, 10); // zero on errors
                if (seconds <= 0 || seconds > std::numeric_limits<time_t>::max())
                    break;
                const auto tm = static_cast<time_t>(seconds);
```